### PR TITLE
[17.09] Fail job if tools that use galaxy.json write to stderr

### LINKED ISSUE #5214: errors in upload are not shown as erroneous dataset
Errors happening during download don't mark the created data set as error.

screenshot from 2017-12-14 13-19-40

### PR DESCRIPTION
**Pull request by mvdbeek**
This should fix #5214.

### CODE REVIEW

**Comment by mvdbeek** [comment]
This integration test fails now: https://jenkins.galaxyproject.org/job/docker-integration/3330/testReport/integration.test_upload_configuration_options/DefaultBinaryContentFiltersTestCase/test_gzipped_html_content_blocked_by_default/ ... I guess it expects a dataset in the OK state, but I'd argue failing it is better, right ?

File: test/api/test_tools.py [hunk 1]
```diff
@@ -1407,6 +1416,19 @@ def _run(self, tool_id, history_id, inputs, assert_ok=False, tool_version=None):
         else:
             return create_response
 
+    def _upload_from_url(self, url):
```
Comment (jmchilton) [inline test/api/test_tools.py:1419]: In this same directory there is a test_tools_upload.py test case filled with tons of upload tests - can you move these there?

File: test/integration/test_upload_configuration_options.py [hunk 2]
```diff
@@ -82,9 +82,9 @@ def test_random_binary_allowed(self):
 
     def test_gzipped_html_content_blocked_by_default(self):
         dataset = self.dataset_populator.new_dataset(
-            self.history_id, 'file://%s/bad.html.gz' % TEST_DATA_DIRECTORY, file_type="auto", wait=True
+            self.history_id, 'file://%s/bad.html.gz' % TEST_DATA_DIRECTORY, file_type="auto", wait=True, assert_ok=False
         )
-        dataset = self.dataset_populator.get_history_dataset_details(self.history_id, dataset=dataset)
+        dataset = self.dataset_populator.get_history_dataset_details(self.history_id, dataset=dataset, assert_ok=False)
```
Comment (jmchilton) [inline test/integration/test_upload_configuration_options.py:87]: Yes - this is good. Thanks!

File: test/api/test_tools.py (hunk 1 above)
Comment (mvdbeek) [reply test/api/test_tools.py:1419]: Hmm, that's only in dev. Should I create the file ?

Comment (jmchilton) [reply test/api/test_tools.py:1419]: Doh - good point. Lets keep it as is - I'll try to remember to move things around after we merge this forward.

**Comment by jmchilton** [comment]
Any interest in rearranging this so the tool directly just declares the output in question is in error? (https://github.com/mvdbeek/galaxy/pull/6).

File: test/api/test_tools.py [hunk 3]
```diff
@@ -1366,6 +1366,15 @@ def test_combined_mapping_and_subcollection_mapping(self):
         }
         self._check_combined_mapping_and_subcollection_mapping(history_id, inputs)
 
+    def test_upload_from_invalid_url(self):
+        history_id, dataset_id = self._upload_from_url('https://usegalaxy.org/bla123')
+        datset_details = self.dataset_populator.get_history_dataset_details(history_id, dataset_id=dataset_id, assert_ok=False)
```
Comment (nsoranzo) [inline test/api/test_tools.py]: s/datset_details/dataset_details/ 

Also on the line below.

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**